.github/workflows/publish.yml

```
@@ -0,0 +1,52 @@
+name: publish
+on:
+  push:
+    tags:
+      - 'v.*'
```

I don't think github uses actual regex there so it doesn't have the effect you. Instead it's glob pattern. You can use:
```suggestion
      - "v[0-9]+.[0-9]+.[0-9]+"
      - "v[0-9]+.[0-9]+.[0-9]+-rc[0-9]+"
```
I know that that pattern works

---

It is better to have them as different workflows so that you can trigger one from another. Modern PyPI without token requires to select a trusted workflow file, so that's also an incentive to have such a separation.

---

Done at https://github.com/mwouts/jupytext/pull/1084/commits/a71f3360061ee14224c0fa80fcb7e0f037841ec4